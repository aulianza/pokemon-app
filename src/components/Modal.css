.modal {
    position: fixed;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.5);
    display: flex;
    align-items: center;
    justify-content: center;
    z-index: 1;
    transition: all 0.3s ease-in-out;
}

.modal-content {
    width: 500px;
    border-radius: 8px;
    background-color: #fff;
    transition: all 0.3s ease-in-out;
}

.modal-header,
.modal-footer {
    padding: 10px;
}

.modal-title {
    margin: 0;
    font-size: 1.2em;
    font-weight: 500;
    padding: .5rem;
}

.modal-body {
    padding: 10px;
}
