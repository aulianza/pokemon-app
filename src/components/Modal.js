import React from "react";

import "./Modal.css";

const Modal = (props) => {
    return (
        <div className="modal">
            <div className="modal-content">
                <div className="modal-header">
                    <div className="modal-title">{props.title}</div>
                </div>
                <div className="modal-body">{props.children}</div>
                <div className="modal-footer"></div>
            </div>
        </div>
    );
};

export default Modal;
