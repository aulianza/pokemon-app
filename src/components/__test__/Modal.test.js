import Modal from "../Modal";
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("modal-component, snapshot", () => {
    act(() => {
        render(<Modal />, container)
    })
    expect(container).toMatchSnapshot();
});
