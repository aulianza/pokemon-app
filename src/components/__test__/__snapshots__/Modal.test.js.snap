// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal-component, snapshot 1`] = `
<div>
  <div
    class="css-1m921on"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <div
          class="modal-title"
        />
      </div>
      <div
        class="modal-body"
      />
      <div
        class="modal-footer"
      />
    </div>
  </div>
</div>
`;
